app/storage/metadata_parser.py

```diff
@@ -99,11 +111,11 @@ def _validate_metadata_values_are_valid(claims, required_metadata):
             name = metadata_field['name']
             claim = claims.get(name)
             if claim:
-                VALIDATORS[metadata_field['validator']](claim)
                 logger.debug('parsing metadata', key=name, value=claim)
+                VALIDATORS[metadata_field['validator']](claim)
 
     except (RuntimeError, ValueError, TypeError) as error:
-        logger.error('unable to parse metadata', exc_info=error)
+        logger.error('Unable to parse metadata for key `{}`'.format(name), exc_info=error)
```

Comment:
We should maintain structured logging format
`logger.error('Unable to parse metadata', key=name, exc_info=error)`